src/pages/api/modules/usenet/history.ts

```diff
@@ -33,23 +34,72 @@ async function Get(req: NextApiRequest, res: NextApiResponse) {
       throw new Error(`Service with ID "${req.query.serviceId}" could not be found.`);
     }
 
-    if (!service.apiKey) {
-      throw new Error(`API Key for service "${service.name}" is missing`);
+    let response: UsenetHistoryResponse;
+    switch (service.type) {
+      case 'NZBGet': {
+        const url = new URL(service.url);
+        const options = {
+          host: url.hostname,
+          port: url.port,
+          login: service.username,
+          hash: service.password,
+        };
+
+        const nzbGet = new NZBGet(options);
+
+        const nzbgetHistory:[] = await new Promise((resolve, reject) => {
+          nzbGet.history(false, (err: any, result: any) => {
+            if (!err) {
+              resolve(result);
+            } else {
+              reject(err);
+            }
+          });
+        });
+
+        if (!nzbgetHistory) {
+          throw new Error('Error while getting NZBGet history');
+        }
+
+        const nzbgetItems: UsenetHistoryItem[] = nzbgetHistory.map((item: any) => ({
```

Comment:
Replace type ``any`` with a type definition

Reply:
I am working on getting types created for this (and other return types from the nzbget api)